libraries/poco/md5/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(MD5Encrypter)
+
+if(CMAKE_VERSION VERSION_LESS 3.0.0)
+    include(CheckCXXCompilerFlag)
+    check_cxx_compiler_flag(-std=c++11 COMPILER_SUPPORTS_CXX11)
+    check_cxx_compiler_flag(-std=c++0x COMPILER_SUPPORTS_CXX0X)
+    if(COMPILER_SUPPORTS_CXX11)
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
```

Comment:
it would be nice to use based on target instead:
`set_target_properties(md5 PROPERTIES CXX_STANDARD 11)`

https://cmake.org/cmake/help/v2.8.12/cmake.html#command:set_target_properties

Reply:
cc/ @SSE4

Reply:
I don't have objections, let's use `set_target_properties`

Reply:
...but I thought that this property was introduced in `CMake >= v3.1`...

Reply:
I think comment applies to the `set(CMAKE_CXX_STANDARD 11)` part, where we can safely change to `set_target_properties`

Reply:
> I think comment applies to the set(CMAKE_CXX_STANDARD 11) part, where we can safely change to set_target_properties

Yes.

> ...but I thought that this property was introduced in CMake >= v3.1...

Me too. That's why for many times I used `set_property` instead. But according the documentation it's supported since 2.8 :thinking:

Reply:
The variable CXX_STANDARD was not defined until CMake 3.1, I think.

Reply:
In any case, please recall the objective of this is to support the documentation, and the support of the documentation is not to be 100% perfect, but to explain in the most concise and clear way how Conan works. Having in the docs 14 extra lines to set the C++ standard correctly doesn't help to explain the conanbuildinfo.cmake include and conan_basic_setup(), which is what is being explained. Adding all those lines only dilute the information and distract the reader.

Reply:
I agree, but when I tried to compile the example locally during testing of CLion plug-in, I was getting errors on unrecognized argument `-std=c++11`. so, if possible, I'd prefer to have examples working out of the box.

Reply:
Will use a simplified version of the cmakelists in the docs and indicate it explicitly

Reply:
I have never seen this error, only warnings when it does not apply. But if it is an error, yes, it should be fixed so it work in most platforms. In which platform, compiler, and version, cmake, etc... did you get that error?

Reply:
Visual Studio 2019 emits warnings D9002, which fails with "treat warnings as errors"
also old GCC version (e.g. 4.6 on CentOS 6) wants `-std=c++0x`)